### DISCUSSION TITLE
.NET Framework error - This application could not be started.

### DISCUSSION THREAD

**Discussion by risbi0** [simeononsecurity/Windows-Optimize-Harden-Debloat#79] [Q&A] [answered]
I tried to push my commits to GitHub but got this [error](https://answers.microsoft.com/en-us/windows/forum/all/error-this-application-could-not-be-started/5e2a95ae-f5ba-46a8-bb27-77359dc1b5c9) message but for GitHub.UI.exe. I've installed the latest version of .NET like it said but it still displays the same error when I try to push again. Any help?

**Comment by simeononsecurity** [comment]
So you're using github desktop @risbi0 ?
What configuration did you use?

Automated, GUI, Chocolatey, or Manual?
If applicable, what configuration parameters did you use?

We have people using the UI version of github desktop all the time. So it'll likely have to do with an existing configuration on your system that, when combined with ours, cause the issues you're having. If you're using the GitHub ui with any personal repo or system it is possible that whatever authentication/hashing/encryption enabled on that repo is insecure. Our script is rather minimal with it's .net configuration compared to what it could be. But it does disable some really old hashing and encryption algorithms. 

Please clarify the issues you're having. Perhaps try running one of the other install methods. Make sure to reboot your computer and try again.

**Reply by risbi0** [reply]
I used the automated method. Is there a way of undoing all of what the script did?

**Reply by simeononsecurity** [reply]
The script takes a restore point before running, you can use that to restore back.
The automated method is the most tested version. We use the github desktop ui as well. So we can insure you that that is tested and we're sure it's a configuration on your system that is the issue.
However we would love to help you troubleshoot further if you can provide any more details.

**Reply by risbi0** [reply]
I tried to use system restore but it seems that it failed to undo what the script has done. I went to the restore options again and it now has 2 options instead of 1 when I used system restore earlier. It has the same description ("Windows Modules Installer") and type ("Install") but are 8 minutes apart. The date has also the same date from the restore that I used but I don't recall the exact time the restore point had. I took the earlier version and it still doesn't undo it. Is this intended behavior? Any fixes here?

**Reply by simeononsecurity** [reply, marked as answer]
If it failed, likely something else is wrong with your system, unrelated to the script. We've tested this feature thoroughly. @risbi0 
The best course of action is to reinstall windows.
https://simeononsecurity.com/articles/how-to-install-windows-10-and-windows-11-from-scratch/
